crates/agent-sql/src/publications.rs

```diff
@@ -241,28 +241,36 @@ pub async fn resolve_expanded_rows(
             from seeds s join live_spec_flows e
             on s.id = e.source_id and e.flow_type = 'collection'
         ),
-        -- Expand seed collections, captures, and materializations through
-        -- edges that connect captures and materializations to their bound
-        -- collections:
-        --   * A seed capture expands to all bound collections.
-        --   * A seed materialization expands to all bound collections.
-        --   * A seed collection expands to captures or materializations which bind it,
-        --      and from there to all of its other bound collections.
+        -- Expand seed collections to include bound captures & materializations.
```

Comment:
This seems ready for a deeper refactor to make it more readable and verifiable. The `pass_*` naming (which I started tbf) is not helping.

What about introducing `bound_captures` & `bound_materializations`?
 * `bound_captures` is sources which have a target in `seeds` with flow_type `capture`.
 * `bound_materializations` is targets which have a source in `seeds` with flow_type `materialization`.

Then, a second stage `bound_collections` would be the union of:
 * IDs which have a target in `bound_captures` union `seeds` with flow_type `capture`.
 * IDs which have a source in `bound_materializations` union `seeds` with flow_type `materialization`.
 * IDs which have a source in `seeds` with flow_type `collection` (this is current `pass_one_a`).

Then, `pass_two` becomes `backprop_derivations_and_tests` or similar and uses `bound_collections` as its base-case.

Reply:
I took a shot at refactoring along the lines you suggested. I do think it is more clear now. I found that I needed to union the `bound_captures`, `bound_materializations`, `bound_captures`, and `seeds` together for the `backprop_derivations_and_tests` base case. The non-recursive expansions don't include their initial elements, so they have to be added back in somehow before doing the recursive expansion.